### DISCUSSION TITLE
how to create extension class in protocol folder?

### DISCUSSION THREAD

**Discussion by Frank1016** [serverpod/serverpod#1571] [Q&A] [answered]
I would like to create a UserScope class that extends the Scope class that's already defined in Serverpod on the server side, and make it also available in Flutter app side. How to achieve this?

**Comment by jason-downs** [comment]
I'm running into a similar problem, where I'm extending a yaml-defined class as mentioned in the docs https://docs.serverpod.dev/next/concepts/models#custom-methods on the sever side but want that custom extension method to also be available on the flutter app side and not sure the best way to go about that without modifying the generated files.

**Comment by SandPod** [comment, marked as answer]
Hey @Frank1016! 👋 

I think this question was answered here: https://github.com/serverpod/serverpod/discussions/1716

TLDR: Sharing code on models between client and server is complex currently but we want to make is simpler.
